packages/core/src/v3/runEngineWorker/supervisor/session.ts

```diff
@@ -181,7 +184,7 @@ export class SupervisorSession extends EventEmitter<WorkerEvents> {
 
     if (this.queueConsumerEnabled) {
       console.log("[SupervisorSession] Queue consumer enabled");
-      this.queueConsumer.start();
+      await Promise.allSettled(this.queueConsumers.map(async (q) => q.start()));
```

Comment:
This is a good improvement. For the next version we should look at what @ericallam did in the RedisWorker using plimit. It think it doesn't suffer from the issue where one can hold up the entire loop